pkg/server/singleprocess/service_entrypoint.go

```diff
@@ -315,32 +368,52 @@ func (s *Service) EntrypointExecStream(
 	// it's supported first.
 	iexec, ok := s.state(ctx).(serverstate.InstanceExecHandler)
 	if !ok {
-		return status.Errorf(codes.Unimplemented,
-			"state storage doesn't support exec streaming")
+		return hcerr.Externalize(
+			log,
+			status.Errorf(codes.Unimplemented,
+				"state storage doesn't support exec streaming"),
+			"state storage doesn't support exec streaming",
+		)
 	}
 
 	// Receive our opening message so we can determine the exec stream.
 	req, err := server.Recv()
 	if err != nil {
-		return err
+		return hcerr.Externalize(
+			log,
+			err,
+			"failed to receive entrypoint exex stream",
+		)
 	}
 	open, ok := req.Event.(*pb.EntrypointExecRequest_Open_)
 	if !ok {
-		return status.Errorf(codes.FailedPrecondition,
-			"first message must be open type")
+		return hcerr.Externalize(
+			log,
+			status.Errorf(codes.FailedPrecondition,
+				"error reading entrypoint exec stream, first message must be open type"),
+			"error reading entrypoint exec stream, first message must be open type",
+		)
 	}
 
 	// Get our instance and look for this exec index
 	exec, err := iexec.InstanceExecConnect(ctx, open.Open.Index)
 	if err != nil {
-		return err
+		return hcerr.Externalize(
+			log,
+			err,
+			"failed to connect to instance for exec",
+		)
 	}
 	log = log.With("instance_id", exec.InstanceId, "index", open.Open.Index)
 
 	// Mark we're connected
 	if !atomic.CompareAndSwapUint32(&exec.Connected, 0, 1) {
-		return status.Errorf(codes.FailedPrecondition,
-			"exec session is already open for this index")
+		return hcerr.Externalize(
+			log,
+			status.Errorf(codes.FailedPrecondition,
```

Comment:
I think it's ok to return the status.Errorf here because we're not wrapping any error from deeper in the codebase - we already have total control over what it looks like, so we don't really need the `externalize` here.

But i'm ok with adding externalize for consistency.

Reply:
👍 I think I'll leave it for now but that is something I noticed later on (not wrapping errors) and I've started to not convert those for the reason(s) you point out.